kratos/solving_strategies/convergencecriterias/mixed_generic_criteria.h

```diff
@@ -0,0 +1,446 @@
+        const double zero_tol = 1.0e-12;
+        for(int i = 0; i < mVariableSize; i++) {
+            if (global_solution_norms_vector[i] < zero_tol) {
+                global_solution_norms_vector[i] = 1.0;
+            }
+        }
+
+        // Calculate the norm values
+        std::vector<TDataType> var_ratio(mVariableSize, 0.0);
+        std::vector<TDataType> var_abs(mVariableSize, 0.0);
+        for(int i = 0; i < mVariableSize; i++) {
+            var_ratio[i] = std::sqrt(global_increase_norms_vector[i] / global_solution_norms_vector[i]);
+            var_abs[i] = std::sqrt(global_increase_norms_vector[i]) / static_cast<TDataType>(global_dofs_count[i]);
+        }
+
+        // Output the ratio and absolute norms as a tuple
+        return std::make_tuple(var_ratio, var_abs);
+    }
+
+    /**
+     * @brief Method to output the convergence status
+     * This method prints the convergence status to the screen for each one of the checked variables
+     * @param rConvergenceNorms Tuple containing the absolute and relative convergence values
+     */
+    void OutputConvergenceStatus(const std::tuple<std::vector<TDataType>, std::vector<TDataType>>& rConvergenceNorms)
+    {
+        const auto& var_ratio = std::get<0>(rConvergenceNorms);
+        const auto& var_abs = std::get<1>(rConvergenceNorms);
+
+        if (this->GetEchoLevel() > 0) {
+            std::ostringstream stringbuf;
+            stringbuf << "CONVERGENCE CHECK:\n";
+            for(int i = 0; i < mVariableSize; i++) {
+                const auto r_var_data = mVariableDataVector[i];
+                const int key_map = mLocalKeyMap[r_var_data->Key()];
+                stringbuf << " " << r_var_data->Name() << " : ratio = " << var_ratio[key_map] << "; exp.ratio = " << mRatioToleranceVector[key_map] << " abs = " << var_abs[key_map] << " exp.abs = " << mAbsToleranceVector[key_map] << "\n";
+            }
+            KRATOS_INFO("") << stringbuf.str();
+        }
+    }
+
+    /**
+     * @brief Method to check convergence
+     * This method checks the convergence of the provided norms with the user-defined tolerances
+     * @param rConvergenceNorms Tuple containing the absolute and relative convergence values
+     * @return true Convergence is satisfied
+     * @return false Convergence is not satisfied
+     */
+    bool CheckConvergence(const std::tuple<std::vector<TDataType>, std::vector<TDataType>>& rConvergenceNorms)
+    {
+        int is_converged = 0;
+        const auto& var_ratio = std::get<0>(rConvergenceNorms);
+        const auto& var_abs = std::get<1>(rConvergenceNorms);
+
+        for (int i = 0; i < mVariableSize; i++) {
+            const auto r_var_data = mVariableDataVector[i];
+            const int key_map = mLocalKeyMap[r_var_data->Key()];
+            is_converged += var_ratio[key_map] <= mRatioToleranceVector[key_map] || var_abs[key_map] <= mAbsToleranceVector[key_map];
+        }
+
+        if (is_converged) {
```

Comment:
I think this is the difference with respect VelPrCriteria. Here you are only making sure that 1 variable converged right? I think this should be `is_converged = mVariableSize` or something equivalent

Reply:
You are right. This ensures that one of the checked variables is converged. To ensure all of them are converged the check must be `is_converged = mVariableSize`. I'd change it to keep the old behavior. 

Thanks for spotting! 👍 🥇 😄

Reply:
;)